neofs-lens storage sanity SIGSEGV

```
[root@titan1:~]# ./neofs-lens --config /etc/neofs/storage/config.yml storage sanity
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa3c549]

goroutine 1 [running]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/peapod.(*Peapod).Open(0x0, 0x1)
	github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/peapod/peapod.go:201 +0x29
github.com/nspcc-dev/neofs-node/cmd/neofs-lens/internal/storage.sanityCheck.func2(0xc0002b6ff0)
	github.com/nspcc-dev/neofs-node/cmd/neofs-lens/internal/storage/sanity.go:103 +0xcaf
github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/engine.IterateShards(0xc0002b6820, 0x0, 0xc0001e1ca0)
	github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/engine/config.go:58 +0xbcc
github.com/nspcc-dev/neofs-node/cmd/neofs-lens/internal/storage.sanityCheck(0x12c1e00, {0xc0001633a0?, 0x4?, 0xbe75f3?})
	github.com/nspcc-dev/neofs-node/cmd/neofs-lens/internal/storage/sanity.go:59 +0x12a
github.com/spf13/cobra.(*Command).execute(0x12c1e00, {0xc000163380, 0x2, 0x2})
	github.com/spf13/cobra@v1.8.1/command.go:985 +0xaca
github.com/spf13/cobra.(*Command).ExecuteC(0x12be480)
	github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.8.1/command.go:1041
main.main()
	github.com/nspcc-dev/neofs-node/cmd/neofs-lens/root.go:54 +0x1a
zsh: exit 2     ./neofs-lens --config /etc/neofs/storage/config.yml storage sanity
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -24,6 +24,7 @@ Changelog for NeoFS Node
 - `neofs_node_engine_list_containers_time_bucket` and `neofs_node_engine_exists_time_bucket` metrics (#3014)
 - `neofs_node_engine_list_objects_time_bucket` metric (#3120)
 - The correct role parameter to invocation (#3127)
+- nil pointer error for `storage sanity` command (#3151)
 
 ### Changed
 - Number of cuncurrenly handled notifications from the chain was increased from 10 to 300 for IR (#3068)

--- a/cmd/neofs-lens/internal/storage/sanity.go
+++ b/cmd/neofs-lens/internal/storage/sanity.go
@@ -50,8 +50,12 @@ func sanityCheck(cmd *cobra.Command, _ []string) error {
 	defer func() {
 		for _, sh := range shards {
 			_ = sh.m.Close()
-			_ = sh.p.Close()
-			_ = sh.fsT.Close()
+			if sh.p != nil {
+				_ = sh.p.Close()
+			}
+			if sh.fsT != nil {
+				_ = sh.fsT.Close()
+			}
 		}
 	}()
 
@@ -100,11 +104,15 @@ func sanityCheck(cmd *cobra.Command, _ []string) error {
 		if err := sh.m.Open(true); err != nil {
 			return fmt.Errorf("open metabase: %w", err)
 		}
-		if err := sh.p.Open(true); err != nil {
-			return fmt.Errorf("open peapod: %w", err)
+		if sh.p != nil {
+			if err := sh.p.Open(true); err != nil {
+				return fmt.Errorf("open peapod: %w", err)
+			}
 		}
-		if err := sh.fsT.Open(true); err != nil {
-			return fmt.Errorf("open fstree: %w", err)
+		if sh.fsT != nil {
+			if err := sh.fsT.Open(true); err != nil {
+				return fmt.Errorf("open fstree: %w", err)
+			}
 		}
 
 		// metabase.Open(true) does not set it mode to RO somehow
@@ -115,11 +123,15 @@ func sanityCheck(cmd *cobra.Command, _ []string) error {
 		if err := sh.m.Init(); err != nil {
 			return fmt.Errorf("init metabase: %w", err)
 		}
-		if err := sh.p.Init(); err != nil {
-			return fmt.Errorf("init peapod: %w", err)
+		if sh.p != nil {
+			if err := sh.p.Init(); err != nil {
+				return fmt.Errorf("init peapod: %w", err)
+			}
 		}
-		if err := sh.fsT.Init(); err != nil {
-			return fmt.Errorf("init fstree: %w", err)
+		if sh.fsT != nil {
+			if err := sh.fsT.Init(); err != nil {
+				return fmt.Errorf("init fstree: %w", err)
+			}
 		}
 
 		shards = append(shards, sh)
@@ -195,9 +207,17 @@ func checkShard(cmd *cobra.Command, sh storageShard) (int, error) {
 
 			switch id := string(sid); id {
 			case "":
-				err = checkObject(*header, sh.fsT)
+				if sh.fsT != nil {
+					err = checkObject(*header, sh.fsT)
+				} else {
+					err = fmt.Errorf("incorrect metabase, no FSTree, but object found")
+				}
 			case peapod.Type:
-				err = checkObject(*header, sh.p)
+				if sh.p != nil {
+					err = checkObject(*header, sh.p)
+				} else {
+					err = fmt.Errorf("incorrect metabase, no Peapod, but object found")
+				}
 			default:
 				err = fmt.Errorf("uknown storage ID: %s", id)
 			}

